Extension/.gitignore

```diff
@@ -19,3 +19,17 @@ install.lock
 
 # ignore vscode test
 .vscode-test/
+
+# ignore generated localization file
+**/nls.*.json
+**/*.nls.json
+**/*.nls.*.json
+
+# ignore exported localization xlf directory
+localization-export
```

Comment:
The nls files need to be in the VSIX, but are generated so should not be checked in.  I've since modes the localization-export directory to outside of the Extension directory, so moving this to parent .gitignore.